### ISSUE TITLE
GVim starts slowly with enable_for_buffer()

### ISSUE THREAD

**Issue by Neur1n** [ncm2/ncm2#39]
If you're having trouble with NCM2, Please provide the following information
along with your detailed issue description:

### OS
Windows 10 Pro 1803

### neovim `:version` output
GVim 8.1 with patch 1-116

### Minimal vimrc for reproducing the issue
```vim
set nocompatible
set fileformat=unix
set fileformats=unix,dos
set encoding=utf-8

call plug#begin('E:\ProgramFiles\Vim\vimfiles\plugged')
Plug 'ncm2/ncm2'
Plug 'roxma/nvim-yarp'
Plug 'roxma/vim-hug-neovim-rpc'
call plug#end()

autocmd BufEnter * call ncm2#enable_for_buffer()
```

### neovim `:messages` output when the error occurs
N/A.

### Log files if you think the issue is a bug
```sh
gvim --startuptime log.txt
```
with and without the `autocmd BufEnter * call ncm2#enable_for_buffer()`.

### Additional description
In short, is there a solution to  make this autocmd work without slowing down the startup?

Basically, with `autocmd BufEnter * call ncm2#enable_for_buffer()` in my full [vimrc](https://github.com/Neur1n/dotfiles/tree/master/vim), it could take as much as about 2.6s to start GVim while it takes only about 500ms to start without this autocmd. However, it is necessary to put this autocmd in my vimrc since I usually load a session before working.

Startup timing log files of minimal vimrc are attached below. Thanks in advanced.

[log.txt](https://github.com/ncm2/ncm2/files/2223787/log.txt)
[log_wo.txt](https://github.com/ncm2/ncm2/files/2223788/log_wo.txt)

**Comment by roxma** [comment]
Hmm...

If you don't want ncm2 to start as vim starts.

Try something like

```
    function s:ncm2_start(...)
        call ncm2#enable_for_buffer()
        autocmd BufEnter * call ncm2#enable_for_buffer()
    endfunc
    call timer_start(500, function('s:ncm2_start'))
```

or 

```
autocmd InsertEnter * call ncm2#enable_for_buffer()
```

**Comment by Neur1n** [comment]
It works great so far, thanks a lot.

**Comment by YaLTeR** [comment]
I believe I'm hitting the same issue, with [neovim-gtk](https://github.com/daa84/neovim-gtk), except it straight up hits a timeout and shows an error message with
```vim
autocmd BufEnter * call ncm2#enable_for_buffer()
```
so it's rather critical. Launching nvim in terminal works fine.
```vim
autocmd InsertEnter * call ncm2#enable_for_buffer()
```
seems to work too but it's a workaround.

**Comment by YaLTeR** [comment]
Can this be fixed properly? Even the timer workaround isn't perfect and makes `nvim-gtk` freeze on load every now and then.

**Comment by roxma** [comment]
https://github.com/neovim/neovim/issues/2437#issuecomment-106647415

>  You can disable the check by let g:python_host_skip_check = 1 and let g:python3_host_skip_check = 1

@YaLTeR  Does this setting work for you?

**Comment by YaLTeR** [comment]
Actually after some testing it looks like the occasional freeze I'm currently getting is unrelated to ncm2, and also
```vim
autocmd BufEnter * call ncm2#enable_for_buffer()
```
currently works even without a timeout.

(and setting those two variables you suggested didn't seem to prevent the occasional freeze)
